cmd/agent/app/reporter/grpc/flags.go

```diff
@@ -22,43 +22,36 @@ import (
 )
 
 const (
-	gRPCPrefix             = "reporter.grpc."
-	collectorHostPort      = gRPCPrefix + "host-port"
-	retry                  = gRPCPrefix + "retry.max"
-	defaultMaxRetry        = 3
-	collectorTLS           = gRPCPrefix + "tls"
-	collectorTLSCA         = gRPCPrefix + "tls.ca"
-	collectorTLSServerName = gRPCPrefix + "tls.server-name"
+	gRPCPrefix              = "reporter.grpc."
+	collectorHostPort       = gRPCPrefix + "host-port"
+	retry                   = gRPCPrefix + "retry.max"
+	defaultMaxRetry         = 3
+	collectorTLS            = gRPCPrefix + "tls"
+	collectorTLSCA          = gRPCPrefix + "tls.ca"
+	collectorTLSServerName  = gRPCPrefix + "tls.server-name"
+	collectorResolverTarget = gRPCPrefix + "resolver.target"
 )
 
-// Options Struct to hold configurations
-type Options struct {
-	// CollectorHostPort is list of host:port Jaeger Collectors.
-	CollectorHostPort []string
-	MaxRetry          uint
-	TLS               bool
-	TLSCA             string
-	TLSServerName     string
-}
-
 // AddFlags adds flags for Options.
 func AddFlags(flags *flag.FlagSet) {
 	flags.String(collectorHostPort, "", "Comma-separated string representing host:port of a static list of collectors to connect to directly.")
 	flags.Uint(retry, defaultMaxRetry, "Sets the maximum number of retries for a call.")
 	flags.Bool(collectorTLS, false, "Enable TLS.")
 	flags.String(collectorTLSCA, "", "Path to a TLS CA file. (default use the systems truststore)")
 	flags.String(collectorTLSServerName, "", "Override the TLS server name.")
+	flags.String(collectorResolverTarget, "", "Path to collector resolver")
```

Comment:
what does this mean? Would it be the UNS name in our case? If so, should we just reuse the collectorHostPort?

Reply:
yes, i will refactor shortly